src/ui/backend/canvas/ViewBacking.js

```diff
@@ -20,47 +20,48 @@ let exports = {};
  *
  * Models the style object of the canvas View.
  */
-import strPad from '../strPad';
 import BaseBacking from '../BaseBacking';
+import Matrix2D from '../../../platforms/browser/webgl/Matrix2D';
 
-var IDENTITY_MATRIX = {
-  a: 1,
-  b: 0,
-  c: 0,
-  d: 1,
-  tx: 0,
-  ty: 0
-};
+var IDENTITY_MATRIX = new Matrix2D();
 var sin = Math.sin;
 var cos = Math.cos;
 
 var ADD_COUNTER = 900000;
 
+function compareZOrder (a, b) {
+  var zIndexCmp = a._zIndex - b._zIndex;
+  if (zIndexCmp !== 0) {
+    return zIndexCmp;
+  }
+
+  return a._addedAt - b._addedAt;
+}
+
 exports = class extends BaseBacking {
 
   constructor (view) {
     super();
 
-    this._globalTransform = {
-      a: 1,
-      b: 0,
-      c: 0,
-      d: 1,
-      tx: 0,
-      ty: 0
-    };
+    this._globalTransform = new Matrix2D();
     this._cachedRotation = 0;
     this._cachedSin = 0;
     this._cachedCos = 1;
     this._globalOpacity = 1;
     this._view = view;
     this._superview = null;
+
+    this._needsSort = false;
+    this._visiblesNeedsSort = false;
```

Comment:
Should be `this._visiblesNeedSort` since visibles is plural.

If you'd prefer it to align with `this._needsSort`, we could use something like `this._shouldSort` and `this._shouldSortVisibleSubviews`